### DISCUSSION TITLE
Querying datasettes within python notebooks.

### DISCUSSION THREAD

**Discussion by kindly** [simonw/datasette#1329] [Show and tell]
Thanks for the great project and trying to push the boundries in how data can be easily published.

I have been working on a library called [noteql](https://github.com/OpenDataServices/noteql) which is trying to make SQL queries in notebooks as nice as possible.  Read the README for more info.

So as an experimetal feature I have added a `Datasette` backend to it and it works pretty well!  Here is an example notebook using one of the example datasette instances as the backend.

https://colab.research.google.com/drive/1o5wgDnS_CP4C6HVLxvz0ZdGQUaFhWXw9

Would be great to get some feedback on it. 

I have also been experimenting with notebook generation, to give people a ipynb to download or put in google drive, with the basic notebook setup done.  This could potentially be added as a plugin similar to https://datasette.io/plugins/datasette-export-notebook

**Comment by simonw** [comment]
This is really cool! Love the way it gives you new Jupyter magic for running SQL:

<img width="1122" alt="datasette_example_ipynb_-_Colaboratory" src="https://user-images.githubusercontent.com/9599/120903709-90ff1a00-c5fc-11eb-94b4-31641d7c8a61.png">

The Jinja templating for queries is really neat too.
